src/model_signing/manifest.py

```diff
@@ -245,16 +246,156 @@ def key(self) -> ManifestKey:
         return Shard(self._path, self._start, self._end)
 
 
+class SerializationType(metaclass=abc.ABCMeta):
+    """A description of the serialization process that generated the manifest.
+
+    These should record all the parameters needed to ensure a reproducible
+    serialization. These are used to build a manifest back from the signature in
+    a backwards compatible way. We use these to determine what serialization to
+    use when verifying a signature.
+    """
+
+    @property
+    @abc.abstractmethod
+    def serialization_dict(self) -> dict[str, Any]:
+        """The arguments of the serialization method."""
+
+    @classmethod
+    def from_dict(cls, serialization_dict: dict[str, Any]) -> Self:
```

Comment:
thoughts on `serialization_args` and `from_args` or `serialization_parameters` `from_parameters`? `dict` is conveyed already by the typing, and the class docstring mentions the word parameters

Reply:
Those are better names, thank you!